fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/PreparedCommand.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.OriginStatement;
+import org.apache.doris.qe.PreparedStatementContext;
+import org.apache.doris.qe.StmtExecutor;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Prepared Statement
+ */
+public class PreparedCommand extends Command {
+    private static final Logger LOG = LogManager.getLogger(StmtExecutor.class);
+
+    protected List<Placeholder> params = new ArrayList<>();
+    private final LogicalPlan inner;
+
+    private final String stmtName;
+
+    private final OriginStatement originalStmt;
+
+    /**
+     * constructor
+     * @param name the statement name which represents statement id for prepared statement
+     * @param plan the inner statement
+     * @param placeholders the parameters for this prepared statement
+     * @param originalStmt original statement from StmtExecutor
+     */
+    public PreparedCommand(String name, LogicalPlan plan, List<Placeholder> placeholders,
+                OriginStatement originalStmt) {
+        super(PlanType.UNKNOWN);
+        this.inner = plan;
+        if (placeholders != null) {
+            this.params = placeholders;
+        }
+        this.stmtName = name;
+        this.originalStmt = originalStmt;
+    }
+
+    public String getName() {
+        return stmtName;
+    }
+
+    public List<Placeholder> params() {
+        return params;
+    }
+
+    public int getParamLen() {
+        if (params == null) {
+            return 0;
+        }
+        return params.size();
+    }
+
+    public LogicalPlan getInnerPlan() {
+        return inner;
+    }
+
+    public OriginStatement getOriginalStmt() {
+        return originalStmt;
+    }
+
+    /**
+     * return the labels of paramters
+     */
+    public List<String> getLabels() {
+        List<String> labels = new ArrayList<>();
+        if (params == null) {
+            return labels;
+        }
+        for (Placeholder parameter : params) {
+            labels.add("$" + parameter.getExprId().asInt());
+        }
+        return labels;
+    }
+
+    // register prepared statement with attached statement id
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        List<String> labels = getLabels();
+        // register prepareStmt
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("add prepared statement {}, isBinaryProtocol {}",
+                    stmtName, ctx.getCommand() == MysqlCommand.COM_STMT_PREPARE);
+        }
+        ctx.addPreparedStatementContext(stmtName,
+                new PreparedStatementContext(this, ctx, ctx.getStatementContext(), stmtName));
+        if (ctx.getCommand() == MysqlCommand.COM_STMT_PREPARE) {
+            executor.sendStmtPrepareOK((int) ctx.getStmtId(), labels);
+        }
```

Comment:
```suggestion
        executor.sendStmtPrepareOK((int) ctx.getStmtId(), labels);
```

Reply:
COM_STMT_PREPARE need to send such packet, otherwise it maybe text protocol